api/dp-openapi.yaml

```diff
@@ -1290,6 +1290,58 @@ paths:
               schema:
                 $ref: '#/components/schemas/round_stats'
 
+  /storage/clients/{clientid}:
+    get:
+      tags:
+      - storage
+      description: Get JSON blob stored by the client
+      parameters:
+      - name: clientid
+        in: path
+        description: Client ID
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: JSON blob
+          content:
+            application/json:
+              schema:
+                type: object
+    put:
+      tags:
+      - storage
+      description: Save JSON blob for the client
+      parameters:
+      - name: clientid
+        in: path
+        description: Client ID
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: JSON blob that was persisted
+          content:
+            application/json:
+              schema:
+                type: object
```

Comment:
The timestamp is arguably just as useless as the JSON blob because it's always `now()`. I was uncertain what a good return value would be so I modeled what we do for PUTs for word lists -- and what the product at my job does for PUTs and PATCHes -- which is to return the object that was actually serialized.

Open to ideas / thoughts on what API clients should "expect" from a successful PUT / PATCH / etc beyond the HTTP status code.

Reply:
> The timestamp is arguably just as useless as the JSON blob because it's always now(). 

Sure as it represents the last modified timestamp from the server's perspective.

However I do think there is a lot more value to returning this compared to the blob that was sent and is known to the client. The timestamp would allow the client to implement some caching (e.g. don't check with the server if the timestamp is fresh enough), lower the bandwidth needed (by return the JSON object on a `get` request if the timestamp doesn't match, useful for mobile devices) or potentially even allow for delta (this would need to switch to some event based logic). Arguably, neither of those scenarios applies here, but those are enabled by returning the timestamp :smile:

> Open to ideas / thoughts on what API clients should "expect" from a successful PUT / PATCH / etc beyond the HTTP status code.

I don't think we need to return anything from the API on success: the return code says if it was successful and we can return an optional error field to return the cause of errors to the client. Those 2 are what a minimal client would expect. We can always add more as we understand our needs more.